DatadogInternal/Sources/NetworkInstrumentation/OpenTelemetry/OTelHTTPHeadersWriter.swift

```diff
@@ -6,41 +6,45 @@
 
 import Foundation
 
-/// The `OTelHTTPHeadersWriter` should be used to inject trace propagation headers to
-/// the network requests send to the backend instrumented with Open Telemetry.
-/// The injected headers conform to [Open Telemetry](https://github.com/openzipkin/b3-propagation) standard.
+/// The `OTelHTTPHeadersWriter` class facilitates the injection of trace propagation headers into network requests
+/// targeted at a backend expecting [B3 propagation format](https://github.com/openzipkin/b3-propagation).
 ///
 /// Usage:
 ///
 ///     var request = URLRequest(...)
 ///
 ///     let writer = OTelHTTPHeadersWriter(injectEncoding: .single)
-///     let span = DatadogTracer.shared().startSpan("network request")
-///     writer.inject(spanContext: span.context)
+///     let span = Tracer.shared().startRootSpan(operationName: "network request")
+///     Tracer.shared().inject(spanContext: span.context, writer: writer)
 ///
 ///     writer.traceHeaderFields.forEach { (field, value) in
 ///         request.setValue(value, forHTTPHeaderField: field)
 ///     }
 ///
 ///     // call span.finish() when the request completes
 ///
-///
 public class OTelHTTPHeadersWriter: TracePropagationHeadersWriter {
```

Comment:
💡 Even with these changes, I'm concerned about the `OTelHTTPHeadersWriter` name. OTel doesn't dictate this or that headers format and using "OTel" definitely does not imply `B3-*` headers. It should be `B3HTTPHeadersWriter` IMO (consistent with `W3CHTTPHeadersWriter` and bare `HTTPHeadersWriter` which uses `x-datadog-*` headers).

I'd be up for deprecating this symbol and renaming it accordingly. WDYT @ganeshnj @maciejburda @maxep ?

Reply:
[Agreed](https://github.com/DataDog/dd-sdk-ios/pull/1057#pullrequestreview-1187895880) :) `B3*` is technically more correct 👍

Reply:
Renamed `OTelHTTPHeaders*` to `B3HTTPHeaders*` with deprecating the prior. See: https://github.com/DataDog/dd-sdk-ios/pull/1415/commits/4a3819c1f79fcb83f6dbaa12aee1d394906848cb